packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -0,0 +1,222 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+import 'package:flutter_goldens_client/client.dart';
+
+// If you are here trying to figure out how to use golden files in the Flutter
+// repo itself, consider reading this wiki page:
+// https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter
+
+// TODO(Piinks): This file will replace ./client.dart when transition to Skia
+// Gold testing is complete
+
+const String _kGoldctlKey = 'GOLDCTL';
+const String _kServiceAccountKey = 'GOLD_SERVICE_ACCOUNT';
+
+/// An extension of the [GoldensClient] class that interfaces with Skia Gold
+/// for golden file testing.
+class SkiaGoldClient extends GoldensClient {
+  SkiaGoldClient({
+    FileSystem fs = const LocalFileSystem(),
+    ProcessManager process = const LocalProcessManager(),
+    Platform platform = const LocalPlatform(),
+  }) : super(
+    fs: fs,
+    process: process,
+    platform: platform,
+  );
+
+  /// The local [Directory] within the [comparisonRoot] for the current test
+  /// context. In this directory, the client will create image and json files
+  /// for the goldctl tool to use.
+  ///
+  /// This is informed by the [FlutterGoldenFileComparator] [basedir]. It cannot
+  /// be null.
+  Directory _workDirectory;
+
+  /// The path to the local [Directory] where the goldctl tool is hosted.
+  ///
+  /// Uses the [platform] environment in this implementation.
+  String get _goldctl => platform.environment[_kGoldctlKey];
+
+  /// The path to the local [Directory] where the service account key is
+  /// hosted.
+  ///
+  /// Uses the [platform] environment in this implementation.
+  String get _serviceAccount => platform.environment[_kServiceAccountKey];
+
+  @override
+  Directory get comparisonRoot => flutterRoot.childDirectory(fs.path.join('bin', 'cache', 'pkg', 'skia_goldens'));
```

Comment:
Rather than putting the generated images in the same folder 'goldens', created a separate one for skia files.